src/internal/FluencePeer.ts

```diff
@@ -0,0 +1,350 @@
+        let peerId;
+        const peerIdOrSeed = config?.peerId;
+        if (!peerIdOrSeed) {
+            peerId = await randomPeerId();
+        } else if (isPeerId(peerIdOrSeed)) {
+            // keep unchanged
+            peerId = peerIdOrSeed;
+        } else {
+            // peerId is string, therefore seed
+            peerId = await peerIdFromEd25519SK(peerIdOrSeed);
+        }
+        this._selfPeerIdFull = peerId;
+
+        await this._initAirInterpreter(config?.avmLogLevel || 'off');
+
+        this._callServiceHandler = makeDefaultClientHandler();
+
+        if (config?.connectTo) {
+            let connectTo;
+            if (Array.isArray(config!.connectTo)) {
+                connectTo = config!.connectTo;
+            } else {
+                connectTo = [config!.connectTo];
+            }
+
+            let theAddress: ConnectionSpec;
+            let fromNode = (connectTo[0] as any).multiaddr;
+            if (fromNode) {
+                theAddress = new MA(fromNode);
+            } else {
+                theAddress = new MA(connectTo[0] as string);
+            }
+
+            await this._connect(theAddress);
+        }
+    }
+
+    /**
+     * Uninitializes the peer: stops all the underltying workflows, stops the Aqua VM
+     * and disconnects from the Fluence network
+     */
+    async uninit() {
+        await this._disconnect();
+        this._callServiceHandler = null;
+    }
+
+    /**
+     * Get the default Fluence peer instance. The default peer is used automatically in all the functions generated
+     * by the Aqua compiler if not specified otherwise.
+     */
+    static get default(): FluencePeer {
+        return this._default;
+    }
+
+    // internal api
+
+    /**
+     * Does not intended to be used manually. Subject to change
+     */
+    get internals() {
+        return {
+            initiateFlow: this._initiateFlow.bind(this),
+            callServiceHandler: this._callServiceHandler,
+        };
+    }
+
+    // private
+
+    private async _initiateFlow(request: RequestFlow): Promise<void> {
+        // setting `relayVariableName` here. If the client is not connected (i.e it is created as local) then there is no relay
+        request.handler.on(loadVariablesService, loadRelayFn, () => {
+            return this._relayPeerId || '';
+        });
+        await request.initState(this._selfPeerIdFull);
+
+        logParticle(log.debug, 'executing local particle', request.getParticle());
+        request.handler.combineWith(this._callServiceHandler);
+        this._requests.set(request.id, request);
+
+        this._processRequest(request);
+    }
+
+    private _callServiceHandler: CallServiceHandler;
+
+    private static _default: FluencePeer = new FluencePeer();
+
+    private _selfPeerIdFull: PeerId;
+    private _requests: Map<string, RequestFlow> = new Map();
+    private _currentRequestId: string | null = null;
+    private _watchDog;
```

Comment:
watchdog is a single word, no reason for camelCase

Reply:
fixed